llvm/test/SYCLLowerIR/printf_generic_as_no_opt.ll

```diff
@@ -0,0 +1,580 @@
+;; This tests replacement of string literal address space for __spirv_ocl_printf
+;; when no optimizations (inlining, constant propagation) have been performed prior
+;; to the pass scheduling.
+
+;; Compiled with the following command (custom build of SYCL Clang with
+;; SYCLMutatePrintfAddrspacePass turned off):
+;; clang++ -fsycl -fsycl-device-only experimental-printf.cpp -S -O0 -D__SYCL_USE_NON_VARIADIC_SPIRV_OCL_PRINTF__
+;;
+;; // experimental-printf.cpp
+;; #include <CL/sycl.hpp>
+;; using namespace sycl;
+;; int main() {
+;;   queue q;
+;;   q.submit([&](handler &cgh) {
+;;     cgh.single_task([=]() {
+;;       ext::oneapi::experimental::printf("String No. %f\n", 1.0f);
+;;       const char *IntFormatString = "String No. %i\n";
+;;       ext::oneapi::experimental::printf(IntFormatString, 2);
+;;       ext::oneapi::experimental::printf(IntFormatString, 3);
+;;     });
+;;   });
+;;   return 0;
+;; }
+
+; RUN: opt < %s --SYCLMutatePrintfAddrspace -S | FileCheck %s
+; RUN: opt < %s --SYCLMutatePrintfAddrspace -S --enable-new-pm=1 | FileCheck %s
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+%"struct.cl::sycl::detail::AssertHappened" = type { i32, [257 x i8], [257 x i8], [129 x i8], i32, i64, i64, i64, i64, i64, i64 }
+%"class.cl::sycl::range" = type { %"class.cl::sycl::detail::array" }
+%"class.cl::sycl::detail::array" = type { [1 x i64] }
+%"class.cl::sycl::id" = type { %"class.cl::sycl::detail::array" }
+%class.anon = type { %"class.cl::sycl::accessor" }
+%"class.cl::sycl::accessor" = type { %"class.cl::sycl::detail::AccessorImplDevice", %union.anon }
+%"class.cl::sycl::detail::AccessorImplDevice" = type { %"class.cl::sycl::id", %"class.cl::sycl::range", %"class.cl::sycl::range" }
+%union.anon = type { %"struct.cl::sycl::detail::AssertHappened" addrspace(1)* }
+%"class.cl::sycl::detail::accessor_common" = type { i8 }
+%class.anon.0 = type { i8 }
```

Comment:
I guess that nothing of this is important for your test, because it is related to assertions handling and `assert` isn't even used in the test.

Therefore, I suggest to cleanup the test file to make it more readable (and hopefully one day we won't see that code unless `assert` is actually used: #4987, #4694).

Note: I understand that manual removal of some LLVM IR means possible human errors in the test and reduces abilities to automatically regenerate the test, but in this particular situation I value readability more than other things I mentioned.

Reply:
Done in 52f7ca738d1350601220bb8ace6787ab950f9fd9 alongside other LIT-related comments